### PR TITLE
WIP: Essi 835 coverpage

### PR DESCRIPTION
**Pull request by jpengst**

### CODE REVIEW

File: app/services/cover_page_generator.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
+class CoverPageGenerator # rubocop:disable Metrics/ClassLength
+  attr_reader :paged_resource
+  # delegate :paged_resource, to: :paged_resource
```
Comment (labradford) [inline app/services/cover_page_generator.rb:3]: Can you delete rather than comment the delegate?

File: app/services/essi/generate_pdf_service.rb [hunk 2]
```diff
@@ -30,17 +30,21 @@ def create_pdf_file_path
...
     def create_tmp_files(pdf)
-      @resource.file_sets.each.with_index(1) do |fs, i|
+      # raise 'test'
+      sorted_files = @resource.file_sets.sort_by{|file| file.date_uploaded}
+      sorted_files.each.with_index(1) do |fs, i|
         Tempfile.create(fs.original_file.file_name.first, dir_path) do |file|
+          page_size = [CoverPageGenerator::LETTER_WIDTH, CoverPageGenerator::LETTER_HEIGHT]
           file.binmode
           file.write(fs.original_file.content)
           # TODO: Fit image onto entire page. Use contstants instead of hard code. (ESSI-935)
```
Comment (labradford) [inline app/services/essi/generate_pdf_service.rb:46]: do you still need this todo?

File: app/services/essi/generate_pdf_service.rb [hunk 3]
```diff
@@ -30,17 +30,21 @@ def create_pdf_file_path
...
     def generate_pdf_document(file_path)
       Prawn::Document.generate(file_path, margin: [0, 0, 0, 0]) do |pdf|
+        CoverPageGenerator.new(@resource).apply(pdf)
         create_tmp_files(pdf)
       end
     end
 
     def create_tmp_files(pdf)
-      @resource.file_sets.each.with_index(1) do |fs, i|
+      # raise 'test'
```
Comment (labradford) [inline app/services/essi/generate_pdf_service.rb:39]: remove debugger

File: app/services/cover_page_generator.rb [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
+
+  private
+
+  def rights_statement_label(statement)
+    Hyrax::RightsStatementService.new.label(statement)
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  def display_text(prawn_document, text)
```
Comment (ShanaLMoore) [inline app/services/cover_page_generator.rb:110]: @jpengst this method can be deleted, I don't think it's being used.

**Pull request merged**